utils.go

```diff
@@ -127,6 +135,21 @@ func monitorShutdownSignal() {
 					}
 
 				}
+				// All shutdown callbacks ensure that the server is safely terminated
+				// and any concurrent process could be started again
+				if signal == shutdownRestart {
+					path := os.Args[0]
+					cmdArgs := os.Args[1:]
+					cmd := exec.Command(path, cmdArgs...)
```

Comment:
> you need to persist the env for the existing process. Does this support seamless migration of env's, pwd etc?

Also when the browser sends the signal how seamless does JSON-RPC calls work? don't see any explanation in the patch what is the preceding nature. Is this a precursor to 'minio control <command>' ?